src/main/java/com/openlattice/chronicle/controllers/ChronicleStudyController.java

```diff
@@ -130,15 +131,45 @@ public Boolean isKnownParticipant(
         return chronicleService.isKnownParticipant( studyId, participantId );
     }
 
+    @RequestMapping(
+            path = STUDY_ID_PATH + PARTICIPANT_ID_PATH,
+            method = RequestMethod.DELETE
+    )
     @Override
-    public void deleteParticipantAndAllNeighbors(
-            UUID studyId, String participantId, DeleteType deleteType ) {
-        // TODO implement this
+    public Void deleteParticipantAndAllNeighbors(
+            @PathVariable( STUDY_ID ) UUID studyId,
+            @PathVariable( PARTICIPANT_ID ) String participantId,
+            @RequestParam( TYPE ) DeleteType deleteType
+    ) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if ( authentication instanceof JwtAuthentication ) {
+            String token = ( (JwtAuthentication) authentication ).getToken();
+            chronicleService.deleteParticipantAndAllNeighbors( studyId, participantId, deleteType, token );
+        } else {
+            throw new InsufficientAuthenticationException( "request is not authenticated" );
+        }
+
+        return null;
```

Comment:
`return null` is an unreachable statement. Your IDE should make this very clear, and if it doesn't, check your setup.

<img width="696" alt="Screen Shot 2020-06-10 at 4 42 00 PM" src="https://user-images.githubusercontent.com/440299/84329331-51146c80-ab39-11ea-906d-0a529ad583a8.png">

To fix, remove the `else` case and just throw:

```
if ( authentication instanceof JwtAuthentication ) {
  ...
}
throw new InsufficientAuthenticationException( "request is not authenticated" );
```

Reply:
I also prefer the pattern you're suggesting above but the return null is required and not unreachable in the current code here. I think your compiler configuration actually might be broken in some weird way @UnsungHero97

Reply:
I strongly disagree. How is it possible to reach the `return null;` here?

```
if () {
  return ...;
}
else {
  throw ...;
}
return null;
```

Reply:
🤦‍♂️ this is the delete path, ignore me.